June 2019 News (#13506)
idaholab/moose: modules/doc/content/newsletter/2019_06.md

A new type of numerical stabilization has been added to PorousFlow to complement the existing "full upwind" and "no stabilization" choices.  The aim is to reduce numerical diffusion.  The code is based on the theoretical work of Kuzmin and Turek.  Users are strongly encouraged to experiment with this new feature so we can collectively learn the strengths and weaknesses in real simulations.  Only [minor changes to input files](porous_flow/kt.md) are needed to activate the new stabilization.  The numerical stabilization [lead page](porous_flow/stabilization.md) lists pros and cons of various stabilization schemes and provides links to pages that explain and quantify numerical diffusion.

## Stochastic Tools "Batch" Mode

The [Stochastic Tools module](stochastic_tools/index.md) now includes the ability to run
full and transient solves using "batch" mode, which is useful when a MultiApp contains
a large number of sub-applications. In batch mode the MultiApp creates a single sub-application
and re-uses it to perform the multiple simulations, allowing for reduced memory usage when
compared to the default behavior. For more information please refer to [batch_mode.md].

## PETSc Upgrade

> PETSc is upgraded to 3.10.5 for moose users, and the new version provides a better performance.

Author: @fdkong Is the new version of PETSc available in the downloadable package? If so, this should just point people to download a new package. If not, then this section should be removed because building PETSc manually is not something many people do, so this will just be confusing.

We encourage MOOSE users to use this new version. To use this version of PETSc, follow
instructions on [moose website](getting_started/index.md) to update the environment package.

## Conservative Transfer

Conservative transfers have been added to MOOSE to support transfer of a physics quantity
in a conservative way. Conservative transfers currently work with `MultiAppMeshFunctionTransfer`,
`MultiAppInterpolationTransfer`, `MultiAppNearestNodeTransfer` and `MultiAppUserObjectTransfer`.

## Vector Support in InterfaceKernel

The `InterfaceKernel` system has been updated to add compatibility with vector finite
elements. Usage of `InterfaceKernel` for standard finite element variables remains the same as
before, but `VectorInterfaceKernel` is now available for vector finite element variables. Examples
of usage for `VectorInterfaceKernel` with `NEDELEC_ONE` variables in 2D and 3D can be found in the
following two input files:

- [test/tests/interfacekernels/2d_interface/vector_2d.i]
- [test/tests/interfacekernels/3d_interface/vector_3d.i]

## Simplified Module Makefile

> MOOSE now contains 18 physics modules several of which depend on one and other. The "make" system

Reviewer: There should be a comma after "modules", and the phrase "depend on one and other" I think was meant to be "depend on one another". Better would be "depend on each other".

> MOOSE now contains 18 physics modules several of which depend on one and other. The "make" system

Reviewer: ```suggestion
MOOSE now contains 18 physics modules, several of which depend on each other. The "make" system
```

has been simplified at the module level so dependencies no longer need to appear in the Makefile.
Dependencies are listed in a single place (e.g. modules.mk).

The relevant section for every module has been abbreviated. Here is the relevant section for the
phase_field Makefile:

```
# Module
MODULE_NAME := phase_field
include $(MODULE_DIR)/modules.mk
```

## Improved Dynamic Loading Error Messages

MOOSE has long had the capability to load objects from dynamic libraries. The problem is that
when a failure occurs the error message states that the file cannot be opened. This is rarely

> the case so the error is misleading. An improved error message now differentiates between

Reviewer: comma after "case"

> the case so the error is misleading. An improved error message now differentiates between

Reviewer: ```suggestion
the case, so the error is misleading. An improved error message now differentiates between
```
